File: pages/LoginPage.js
class LoginPage {
    constructor(page) {
        this.page = page;
        this.emailPhoneInput = 'input[type="text"], input[type="email"], input[type="tel"]';
        this.passwordInput = 'input[type="password"]';
        this.loginButton = 'button[type="submit"]';
        this.userEmail = 'aghasherazi355@gmail.com';
    }

    async navigateToLoginPage() {
        await this.page.goto('https://app.demo.smartbenefits.pk', {
            waitUntil: 'networkidle',
            timeout: 60000
        });
    }

    async login(email, password) {
        // Wait for and fill email input
        await this.page.waitForSelector(this.emailPhoneInput, { state: 'visible', timeout: 60000 });
        await this.page.fill(this.emailPhoneInput, email);
        
        // Wait for and fill password input
        await this.page.waitForSelector(this.passwordInput, { state: 'visible', timeout: 60000 });
        await this.page.fill(this.passwordInput, password);
        
        // Click login and wait for navigation
        await Promise.all([
            this.page.waitForNavigation({ waitUntil: 'networkidle', timeout: 60000 }),
            this.page.click(this.loginButton)
        ]);
        
        // Additional wait to ensure dashboard is loaded
        await this.page.waitForLoadState('domcontentloaded', { timeout: 60000 });
    }

    async validateDashboardURL() {
        // Wait for any dashboard elements instead of specific URL
        await this.page.waitForSelector('.dashboard, .main-content, nav', { timeout: 60000 });
    }

    async validateUserEmail() {
        // Try multiple possible selectors where email might appear
        const emailSelectors = [
            `text=${this.userEmail}`,
            `[title*="${this.userEmail}"]`,
            `[aria-label*="${this.userEmail}"]`,
            '.user-info, .profile-info'
        ];
        
        for (const selector of emailSelectors) {
            try {
                await this.page.waitForSelector(selector, { timeout: 20000 });
                return;
            } catch (e) {
                continue;
            }
        }
        throw new Error('Could not find user email on dashboard');
    }
}

module.exports = LoginPage;

File: pages_py/login_page.py
from playwright.sync_api import Page
from login_secrets import ACCOUNT_MANAGER_EMAIL

class LoginPage:
    def __init__(self, page: Page):
        self.page = page
        self.email_phone_input = 'input[type="text"], input[type="email"], input[type="tel"]'
        self.password_input = 'input[type="password"]'
        self.login_button = 'button[type="submit"]'
        self.user_email = ACCOUNT_MANAGER_EMAIL

    def navigate_to_login_page(self, url):
        self.page.goto(url, wait_until='networkidle', timeout=60000)

    def login(self, email, password):
        self.page.wait_for_selector(self.email_phone_input, state='visible', timeout=60000)
        self.page.fill(self.email_phone_input, email)
        self.page.wait_for_selector(self.password_input, state='visible', timeout=60000)
        self.page.fill(self.password_input, password)
        with self.page.expect_navigation(wait_until='networkidle', timeout=60000):
            self.page.click(self.login_button)
        self.page.wait_for_load_state('domcontentloaded', timeout=60000)

    def validate_dashboard_url(self):
        self.page.wait_for_selector('.dashboard, .main-content, nav', timeout=60000)

    def validate_user_email(self):
        email_selectors = [
            f'text={self.user_email}',
            f'[title*="{self.user_email}"]',
            f'[aria-label*="{self.user_email}"]',
            '.user-info, .profile-info'
        ]
        for selector in email_selectors:
            try:
                self.page.wait_for_selector(selector, timeout=20000)
                return
            except Exception:
                continue
        raise Exception('Could not find user email on dashboard')
